### ISSUE TITLE
False negatives with getattr()

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ty#3964]
### Summary

With this code:

```python
from typing import reveal_type


def documentation(target: type[object]) -> None:
    doc = getattr(target, "__doc__", "")
    if doc:
        text = doc.strip()
        reveal_type(text)
        text.nonexistent()
```

Different type checkers behave as follows:

| Expression | mypy | pyright | ty 0.0.58 |
| --- | --- | --- | --- |
| `getattr(target, "__doc__", "")` | `Any \| str` | `Any \| str` | `Any \| Literal[""]` |
| `doc` in the truthy branch | `Any \| str` | `Any \| str` | `Any & ~AlwaysFalsy` |
| `reveal_type(text)` | `Any \| str` | `str \| Any` | `Any` |
| `text.nonexistent()` | error | error | no error |

Now, ty is arguably correct here.
The `getattr` overload returns `Any | _T`. Ty solves the empty-string default
as the literal `Literal[""]`; the truthiness check then removes that only
concrete arm, leaving `Any` and hiding invalid string operations. Mypy and
pyright retain `str` through the branch and catch the error.
But of course, the empty string is always falsy, so ty correctly infers `doc` as Any here (or more precisely, `Any & ~AlwaysFalsy`).

But as a user, I would prefer to see an error here; I know that `__doc__` is probably a string and I'd want to be alerted if I do something that doesn't work on strings.

There's a few ways ty could do better here:

- It could special-case `getattr` calls with known-string attribute names. In this case, that would pick up that `__doc__` is declared as `str | None` in typeshed, so ty could infer `doc` as `str | None | Literal[""]` = `str | None`.
- It could use the default argument to getattr as a signal for the expected type. If the user writes `getattr(obj, attr, default)` and `default` is a literal `str`, they probably expect the attribute to be some kind of str. So it could make sense to infer getattr as returning `Any | str` here (i.e., doing literal promotion on the default argument).
